src/rafimage.cpp

```diff
@@ -296,6 +292,37 @@ namespace Exiv2 {
         io_->read(jpg_img_length, 4);
         long jpg_img_off = Exiv2::getULong((const byte *) jpg_img_offset, bigEndian);
         long jpg_img_len = Exiv2::getULong((const byte *) jpg_img_length, bigEndian);
+        byte cfa_header_offset [4];
+        io_->read(cfa_header_offset, 4);
+        // byte cfa_header_length [4];
+        // io_->read(cfa_header_length, 4);
+        long cfa_hdr_off = Exiv2::getULong((const byte *) cfa_header_offset, bigEndian);
+        // long cfa_hdr_len = Exiv2::getULong((const byte *) cfa_header_length, bigEndian);
+        io_->seek(cfa_hdr_off,BasicIo::beg);
+        byte byte_count[4];
+        io_->read(byte_count, 4);
+        byte byte_tag[2];
+        byte byte_size[2];
+        int32_t count = getLong(byte_count, bigEndian);
+        for (int32_t i = 0; i < count; ++i ) {
+            io_->read(byte_tag, 2);
+            uint16_t tag = getUShort(byte_tag, bigEndian);
+
+            io_->read(byte_size, 2);
+            uint16_t tag_size= getUShort(byte_size, bigEndian);
+
+            if (tag == 0x0100) {
+                byte image_height [2];
+                byte image_width [2];
+                io_->read(image_height, 2);
+                io_->read(image_width, 2);
+                pixelHeight_ = getUShort(image_height, bigEndian);
+                pixelWidth_ = getUShort(image_width, bigEndian);
+                break;
+            } else {
+                io_->seek(tag_size, BasicIo::cur);
```

Comment:
Please add a check that this seek works and doesn't put you outside of the image. Mind that `tag_size` is extracted from the image and thus potentially malicious. If it actually puts you outside of the image then all further reads will return garbage values and potentially result in an infinite loop.